src/Framework/AspNetCoreAnalyzers/src/Analyzers/Infrastructure/RouteUsageCache.cs

```diff
@@ -74,6 +92,15 @@ private RouteUsageCache(Compilation compilation)
                 RoutePattern = tree,
                 UsageContext = usageContext
             };
-        });
+        }
+    }
+
+    private RouteUsageModel? GetAndCache(SyntaxToken syntaxToken, ref LastInspectedStringNode? lastInspectedStringNode, CancellationToken cancellationToken)
+    {
+        var routeUsageModelLoader = new RouteUsageModelLoader(_compilation, lastInspectedStringNode, cancellationToken);
+        var model = _lazyRoutePatterns.GetOrAdd(syntaxToken, routeUsageModelLoader.Load);
+        lastInspectedStringNode = routeUsageModelLoader.LastInspectedStringNode;
```

Comment:
>lastInspectedStringNode = routeUsageModelLoader.LastInspectedStringNode [](http://example.com/codeflow?start=8&length=71)

`LastInspectedStringNode?` is a `struct`. Is it possible for two threads to be writing to the same instance? If so, there is a chance of tearing.

Perhaps `LastInspectedStringNode` should be a `class` instead.

Reply:
It looks like `LastInspectedStringNode` instances are not shared across threads, so this may not be an issue. But it might be worth using a `class` so it's clear there is no potential issue.

Reply:
It's not shared. I'll add a comment that it's ok.